pandas/io/pickle.py

```diff
@@ -171,21 +171,22 @@ def read_pickle(
 
     # 1) try standard library Pickle
     # 2) try pickle_compat (older pandas version) to handle subclass changes
-
-    excs_to_catch = (AttributeError, ImportError, ModuleNotFoundError)
+    # 3) try pickle_compat with latin-1 encoding upon a UnicodeDecodeError
 
     try:
-        with warnings.catch_warnings(record=True):
-            # We want to silence any warnings about, e.g. moved modules.
-            warnings.simplefilter("ignore", Warning)
-            return pickle.load(f)
-    except excs_to_catch:
-        # e.g.
-        #  "No module named 'pandas.core.sparse.series'"
-        #  "Can't get attribute '__nat_unpickle' on <module 'pandas._libs.tslib"
-        return pc.load(f, encoding=None)
+        excs_to_catch = (AttributeError, ImportError, ModuleNotFoundError)
```

Comment:
is there a reason to move this inside the "try"?

Reply:
actually, I didn’t move this to inside the try. It is still outside the same`try` as it was before (semantically).

What I did in order to catch `UnicodeDecodeError` only once was to wrap all that code in an outer `try` block that catches `UnicodeDecodeError` regardless if is thrown by `pickle.load(f)` or by `pc.load(f, encoding=none)`.

So, in summary, it’s not that I moved the `excs_to_catch` to inside the `try`, but that I added a new `try` block just before it.

Reply:
ok thanks